### DISCUSSION TITLE
Add child component to parent component.

### DISCUSSION THREAD

**Discussion by Ted-juh** [RomanKubiak/ctrlr#270] [Q&A]
Like the title says: Is there a way to add a child component like a Label to, in this case, a custom component as its parent?

There's not much or close to no information about "addAndMakeVisible", "addChildComponent", "addChildAndSetId" on the old forum. I know what they do in Juce and how to use it in Juce. But not how they are constructed in Ctrlr. 

Can I even use them in Lua? They are bound to Lua though. 
.def("addAndMakeVisible", (void (Component::*)(Component *, int))&Component::addAndMakeVisible)
.def("addChildComponent", (void (Component::*)(Component *, int))&Component::addChildComponent)

But somehow I fail to get the syntax right on how to construct them in proper Lua. 
Anyone?

**Comment by damiensellier** [comment]
Same here I was wondering what's the right lua syntax to construct/destruct a component. I will be glad to see the answer.

**Comment by Godlike-Productions** [comment]
Only way is to test, I'm afraid. Do a what() on the parent component and see if addAndMakeVisible or addChildComponent are available.
I think you'd probably have better luck with the second (ie both components are already on the canvas)

Try parentComponent:addChildComponent(childComponent).  That would be where I'd start. LUA will tell you if it's not happy.

**Reply by Ted-juh** [reply]
In short, Lua not happy. 

no matching overload found, candidates:
void addAndMakeVisible(Component&,Component&,int)
void addAndMakeVisible(Component&,Component*,int)

As far as I know, "int" is to set a z-order. Should be -1. But the problem is the first part. What do I fill in for "Component&,Component&" or "Component&,Component*"?

**Reply by Godlike-Productions** [reply]
It should be the name of your components. ie panel:getModulatorByName("Modulator-2"):getComponent()
